Tests/Data/Mechanics/Linear/Python/gen-unit-circle.py

```diff
@@ -0,0 +1,154 @@
+writer.Write()
+
+# extract boundary of unit circle
+R_squared = new_coords[:,0]**2 + new_coords[:,1]**2
+phi = np.arctan2(new_coords[:,1], new_coords[:,0])
+
+idcs = np.where(abs(R_squared - 1) < 1e-8)
+idcs = idcs[np.argsort(phi[idcs])]  # sorted with ascending polar angle
+
+
+with open(sys.argv[3], "w") as fh:
+    fh.write("""<?xml version="1.0" encoding="ISO-8859-1"?>
+<?xml-stylesheet type="text/xsl" href="OpenGeoSysGLI.xsl"?>
+
+<OpenGeoSysGLI xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.opengeosys.org/images/xsd/OpenGeoSysGLI.xsd" xmlns:ogs="http://www.opengeosys.org">
+    <name>geom</name>
+    <points>
+        <point id="0" x="0" y="0" z="0" name="center"/>
+        <point id="1" x="0" y="1" z="0" name="top"/>
+        <point id="2" x="1" y="0" z="0"/>
+
+    """
+
+    for i, (x, y, z) in enumerate(new_coords[idcs]):
+        fh.write('        <point id="{}" x="{}" y="{}" z="{}" />\n'.format(
+            i+3, x, y, z)
+
+    fh.write("""</points>
+
+    <polylines>
+        <polyline id="0" name="left">
+            <pnt>0</pnt>
+            <pnt>1</pnt>
+        </polyline>
+        <polyline id="1" name="bottom">
+            <pnt>0</pnt>
+            <pnt>2</pnt>
+        </polyline>
+        <polyline id="2" name="outer">\n""")
+
+    for i in range(len(new_coords)):
+        fh.write("            </pnt>{}</pnt>\n".format(i+3))
+
+    fh.write("""</polyline>
+    </polylines>
+</OpenGeoSysGLI>""")
+
+
+
+
+
+
+sys.exit()
+
+
+if False:
+    # If the input is a regular mesh, there is the possibility to make bins of
+    # points with equal radius. For each radius those points can than be
+    # distributed with even spacing at the quarter circles.
+
+    r2 = np.sqrt(c2[:,0]**2 + c2[:,1]**2)
+    # r2 = c2[:,0]**2
+    alpha2 = np.arctan2(c2[:,1], c2[:,0])
+
+    bins = [ [] for _ in range(CELLS_PER_DIRECTION+1) ]
+    nbin = np.round(r2 * CELLS_PER_DIRECTION).astype(int)
+    for node, b in enumerate(nbin):
+        bins[b].append(node)
+
+    for b in bins:
+        b.sort(key=lambda n: alpha2[n])
+        # print(len(b))
+
+    c3 = np.zeros_like(c2)
+
+    for node, r in enumerate(r2):
+        b = nbin[node]
+        i = bins[b].index(node)
+        if len(bins[b]) == 1:
+            phi = 0.0
+        else:
+            phi = np.pi * 0.5 / (len(bins[b]) - 1) * i
+
+        c3[node, 0] = r * np.cos(phi)
+        c3[node, 1] = r * np.sin(phi)
+
+    c3 = numpy_to_vtk(c3, 1)
+    c3.SetName("coords")
+
+    c2 = numpy_to_vtk(c2, 1)
+    c2.SetName("c2")
+    pts = vtk.vtkPoints()
+    pts.SetData(c3)
+
+    r2 = numpy_to_vtk(r2, 1)
+    r2.SetName("r2")
+
+    a = numpy_to_vtk(a, 1)
+    a.SetName("a")
+    # self.GetOutputDataObject(0).GetPointData().AddArray(a)
+    # self.GetOutputDataObject(0).GetPointData().AddArray(r2)
+    # self.GetOutputDataObject(0).GetPointData().AddArray(c2)
+
+    # self.GetOutputDataObject(0).SetPoints(pts) #.SetData(c2)
```

Comment:
Maybe some debug comments could be removed...

Reply:
removed by now.